pkg/controller/mizar/mizar-arktos-network-controller.go

```diff
@@ -149,60 +157,150 @@ func (c *MizarArktosNetworkController) syncNetwork(eventKeyWithType KeyWithEvent
 		klog.V(4).Infof("Finished syncing service %q (%v)", key, time.Since(startTime))
 	}()
 
+	switch event {
+	case EventType_Create:
+		err := c.processNetworkCreation(key)
+		if err != nil {
+			return err
+		}
+	default:
+		panic(fmt.Sprintf("unimplemented for eventType %v", event))
+	}
+	return nil
+}
+
+func (c *MizarArktosNetworkController) processNetworkCreation(key string) error {
 	tenant, name, err := cache.SplitMetaTenantKey(key)
 	if err != nil {
 		return err
 	}
 
-	net, err := c.netLister.NetworksWithMultiTenancy(tenant).Get(name)
+	network, err := c.networkLister.NetworksWithMultiTenancy(tenant).Get(name)
 	if err != nil {
 		return err
 	}
 
-	klog.Infof("Mizar-Arktos-Network-controller - get network: %#v.", net)
+	//Find out the paths of default template to create vpc and subnet
+	vpc := network.Spec.VPCID
+	//subnet := vpc + subnetSuffix
+	subnet := fmt.Sprintf("%s%s", vpc, subnetSuffix)
+	klog.V(4).Infof("Processing arktos network: %#v. vpc [%v], subnet [%v]. Network Type [%v]", network, vpc, subnet, network.Spec.Type)
 
-	switch event {
-	case EventType_Create:
-		err = c.processNetworkCreation(net, eventKeyWithType)
-	default:
-		panic(fmt.Sprintf("unimplemented for eventType %v", event))
+	//skip update or create if type is not mizar or network status is ready
+	if network.Spec.Type != mizarNetworkType {
+		c.recorder.Eventf(network, corev1.EventTypeNormal, "NotRelevent", "Skip processing non mizar network")
+		return nil
 	}
+
+	// Create default VPC and Subnet
+	err = createVpcAndSubnet(vpc, subnet, c.dynamicClient)
 	if err != nil {
+		klog.Errorf("Create VPC and Subnet failed for tenant %v. Error: (%v).", network.Tenant, err)
 		return err
 	}
-	return nil
-}
-
-func (c *MizarArktosNetworkController) processNetworkCreation(network *v1.Network, eventKeyWithType KeyWithEventType) error {
-	//skip update or create if type is not mizar or network status is ready
-	key := eventKeyWithType.Key
-	if network.Spec.Type != mizarNetworkType || network.Status.Phase == v1.NetworkReady {
-		c.recorder.Eventf(network, corev1.EventTypeNormal, "processNetworkCreation", "Type is not mizar, nothing to be done in mizar cluster: %v.", network)
-		return nil
-	}
 
 	msg := &BuiltinsArktosMessage{
 		Name: network.Name,
 		Vpc:  network.Spec.VPCID,
 	}
 
 	response := c.grpcAdaptor.CreateArktosNetwork(c.grpcHost, msg)
-
-	code := response.Code
-	context := response.Message
-
-	switch code {
+	switch response.Code {
 	case CodeType_OK:
-		klog.Infof("Mizar handled arktos network and vpc id update successfully: %s", key)
+		klog.Infof("Mizar handled arktos network %v successfully", key)
```

Comment:
should have log level.

Reply:
This is one message per VPC. Intended to keep the log.